### PR TITLE
Revise telemetry_server - Add sys_status

### PR DESCRIPTION
**Pull request by bazfp**
- Made publish messages `SYNC`
- Removed unneeded rpcs from the `telemetry_server.proto` 
- Added support for more telemetry message types (`sys_status`, `extended_sys_state`)
- VTOL types
- RC receiver types
~Added Vehicle Arming~ Moved to `action_server` - https://github.com/mavlink/MAVSDK-Proto/pull/244
~Added Flight_modes~ Moved to `action_server` - https://github.com/mavlink/MAVSDK-Proto/pull/244

### CODE REVIEW

File: protos/telemetry_server/telemetry_server.proto [hunk 1]
```diff
@@ -13,37 +13,48 @@ option java_outer_classname = "TelemetryServerProto";
...
-    // Publish to 'HealthAllOk' updates.
-    rpc PublishHealthAllOk(PublishHealthAllOkRequest) returns(PublishHealthAllOkResponse) {}
+    rpc PublishRawImu(PublishRawImuRequest) returns(PublishRawImuResponse) { option (mavsdk.options.async_type) = SYNC; }
     // Publish to 'unix epoch time' updates.
-    rpc PublishUnixEpochTime(PublishUnixEpochTimeRequest) returns(PublishUnixEpochTimeResponse) {}
+    rpc PublishUnixEpochTime(PublishUnixEpochTimeRequest) returns(PublishUnixEpochTimeResponse) { option (mavsdk.options.async_type) = SYNC; }
+    // Subscribe to ARM/DISARM commands
+    rpc SubscribeArmDisarm(SubscribeArmDisarmRequest) returns(stream ArmDisarmResponse) { option (mavsdk.options.async_type) = ASYNC; }
+    // Subscribe to DO_SET_MODE
+    rpc SubscribeDoSetMode(SubscribeDoSetModeRequest) returns(stream DoSetModeResponse) { option (mavsdk.options.async_type) = ASYNC; }
```
Comment (julianoes) [inline protos/telemetry_server/telemetry_server.proto:50]: Oh, I would have expected this to be in an `ActionServer` plugin. Was that not the intention?

File: protos/telemetry_server/telemetry_server.proto [hunk 2]
```diff
@@ -489,6 +589,18 @@ message Imu {
     uint64 timestamp_us = 5; // Timestamp in microseconds
 }
 
+// Arming message type
+message ArmDisarm
+{
+    int32 arm = 1; // Should vehicle arm
+    int32 force = 2; // Should arm override pre-flight checks
```
Comment (julianoes) [inline protos/telemetry_server/telemetry_server.proto:596]: Shouldn't these be booleans?

File: protos/telemetry_server/telemetry_server.proto [hunk 3]
```diff
@@ -188,6 +242,35 @@ message PublishUnixEpochTimeResponse {
     TelemetryServerResult telemetry_server_result = 1;
 }
 
+message ArmDisarmResponse
+{
+    TelemetryServerResult telemetry_server_result = 1;
+    ArmDisarm arm = 2;
```
Comment (julianoes) [inline protos/telemetry_server/telemetry_server.proto:248]: Why is this required as well when it already returns whether the arm or disarm command succeeded or not?

File: protos/telemetry_server/telemetry_server.proto [hunk 4]
```diff
@@ -306,6 +396,16 @@ enum FlightMode {
     FLIGHT_MODE_RATTITUDE = 14; // In 'Rattitude' mode
 }
 
+// Maps to MAV_VTOL_STATE
+enum VTOLState
```
Comment (julianoes) [inline protos/telemetry_server/telemetry_server.proto:400]:
```suggestion
enum VtolState
```

**Review by julianoes — COMMENTED** [review]
I'm a bit confused/concerned about the additional logic in the telemetry server which seems to be more around action rather than telemetry.

File: protos/telemetry_server/telemetry_server.proto (hunk 1 above)
Comment (julianoes) [reply protos/telemetry_server/telemetry_server.proto:50]: Alternatively, you could make an autopilot_server plugin and have it contain both.

File: protos/telemetry_server/telemetry_server.proto (hunk 3 above)
Comment (bazfp) [reply protos/telemetry_server/telemetry_server.proto:248]: You cannot tell if it is an "Arm" or "Disarm" event

**Comment by bazfp** [comment]
I've removed all `action_server` logic to it's own PR and removed some of the publishing which don't map to telemetry plugin (`Health` & `FlightModes` all depend on the heartbeat, which is not handled here)

`telem_server` looking leaner and cleaner.

**Review by julianoes — APPROVED** [review]

**Review by julianoes — APPROVED** [review]
